x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/utils/MlStrings.java

```diff
@@ -33,6 +33,38 @@ public final class MlStrings {
 
     public static final int ID_LENGTH_LIMIT = 64;
 
+    // for testing
+    public static String[] someInvalidChars = {
```

Comment:
Please make this `final`:

```
    public static final String[] SOME_INVALID_CHARS = {
```

Also, please move it to `MlStringsTests`, as it's purely for testing and doesn't need to be built into the production jars.